lib/net/ssh/connection/channel.rb

```diff
@@ -497,6 +496,12 @@ def enqueue_pending_output #:nodoc:
             break
           end
         end
+
+        if @closing and not @sent_close and output.length < 1
```

Comment:
What's the reason for putting this code here, vs process? In process we already seems to have similiar handling for EOF.

https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/connection/channel.rb#L310